Reorganize version articles

BillWagner opened the pull request:
Picking up the work done in #6081 to carry it to completion.

This fixes several issues:
Fixes #4241 
Fixes #4352 
Fixes #5304 
Fixes #6249 

/cc @KathleenDollard 

[Internal review link](https://review.docs.microsoft.com/en-us/dotnet/core/versions/index?branch=pr-en-us-6453) Include pages under this TOC node.

Notes for reviewers:

@jcouv @cartermp I've updated the language versions in versions-history.md based on previous comments. 

@leecow I've updated the discussion on support to point to the official page and removed the duplicate information.  

@leecow The uinstall instructions for Linux and MacOS were based on your earlier information, and working on my machines. apt-get is not installed by default on MacOS, so I explained how to remove versions manually.

KathleenDollard committed:
Reorganize version topic

BillWagner committed:
update per feedback on PR #6081

Moving those changes to a new branch and PR.

BillWagner committed:
interim checkin

switching branches

BillWagner commented on docs/core/versions/version-history.md:
```diff
@@ -0,0 +1,54 @@
+.. Footnote for VB versions aligning to Visauls Studio
+
+
+| Date       | .NET Core SDK(4) | .NET Core Runtimes | Contains C# | Contains VB | Notes |
```
@cartermp Do you know where I can get information for an F# column for this table?

cartermp replied:
We don't have this listed anywhere I think, but the F# language version for all of these is 4.1.

Starting with SDK 2.1.400, it will be F# 4.5.

BillWagner commented on docs/core/versions/version-history.md:
```diff
@@ -0,0 +1,54 @@
+.. Footnote for VB versions aligning to Visauls Studio
+
+
+| Date       | .NET Core SDK(4) | .NET Core Runtimes | Contains C# | Contains VB | Notes |
```
@danroth27 Do you know where I can get source for an ASP.NET Core runtime for this table?

BillWagner committed:
Next draft

Remaining tasks:

- [ ] Finish table in version-history
- [ ] write uinstall instructions, and guidance on which versions you need
- [ ] update LTS information

BillWagner committed:
write uninstall

Moving to Mac to finish uninstall

BillWagner committed:
Add unix and linux uinstall instructions.

BillWagner committed:
proofread

BillWagner committed:
fix a tab error

BillWagner committed:
update Linux uninstall instructions

BillWagner commented:
@leecow I updated the Linux instructions after our discussion. I think they are much better now. Can you review again?

leecow commented on docs/core/versions/remove-runtime-sdk-versions.md:
````diff
@@ -0,0 +1,186 @@
+
+```console
+$ dotnet --list-sdks
+1.0.1 [/usr/local/share/dotnet/sdk]
````
These are all Mac locations. Linux installs will be /usr/share/dotnet.

leecow commented on docs/core/versions/remove-runtime-sdk-versions.md:
```diff
@@ -0,0 +1,186 @@
+If you installed .NET Core using a package manager, you use that same package manager to uninstall .NET SDK or runtime. .NET Core installations support most popular package managers. Consult the documentation for your distribution's package manager for the precise syntax on your environment:
+
+- [apt-get](https://wiki.debian.org/apt-get) is used by Debian based systems, including Ubuntu.
+- [yum](http://yum.baseurl.org/wiki/YumCommands) is used on Fedora, SUSE Linux Enterprise System (SLES), CentOS, and Oracle Linux.
```
SLES uses zypper by default

leecow replied:
Also, it's more idiomatic to reference man pages than web sites. eg dnf(8)

BillWagner replied:
I updated all the references to the online man pages. LMK if you have better links for any of them.

leecow commented on docs/core/versions/remove-runtime-sdk-versions.md:
```diff
@@ -0,0 +1,186 @@
+
+In almost all cases, the command to remove a package is `remove`.
+
+The package name for the .NET Core SDK installation for most package managers is `dotnet-sdk`, followed by the version number. For example, the .NET Core SDK version 2.1 is the package `dotnet-sdk-2.1`.
```
This versioning scheme is only valid for 2.1+ for the runtime and 2.1.300+ for the SDK. Older versions require the entire version number, eg dotnet-sdk-2.1.200.

leecow reviewed (changes requested):
Thanks Bill, couple of notes and comments to address.

BillWagner committed:
respond to feedback on Linux uninstall

JRAlexander commented on docs/core/versions/remove-runtime-sdk-versions.md:
```diff
@@ -0,0 +1,186 @@
+---
+title: How to remove .NET runtime and SDK
+description: Instructions for removing the .NET Core Runtime and SDK components on Windows, Mac, and Linx
```
Linux

JRAlexander commented on docs/core/versions/remove-runtime-sdk-versions.md:
```diff
@@ -0,0 +1,186 @@
+
+Over time, as you install updated versions of the .NET Core runtime and SDK, you may want to remove outdated versions of .NET Core. Removing older versions of the runtime may change the runtime chosen to run shared framework applications, as detailed in the article on [.NET Core version selection](selection.md).
+
+The .NET CLI has options you can use to list the versions of the SDK and runtime that are installed on your machine.  Use [`dotnet --list-sdks`](../tools/dotnet.md#options) to see the list of SDKs installed on your machine. Use [`dotnet --list-runtimes`](../tools/dotnet.md#options) to see the list of runtimes installed on your machine. The following text shows typical output for Windows or unix:
```
unix: -> should this be Linux to match tab heading and output below?

JRAlexander commented on docs/core/versions/remove-runtime-sdk-versions.md:
```diff
@@ -0,0 +1,186 @@
+
+In almost all cases, the command to remove a package is `remove`.
+
+The package name for the .NET Core SDK installation for most package managers is `dotnet-sdk`, followed by the version number. Starting with the version 2.1.300 of the .NET Core SDK and version `2.1` of the runtime, only the major and minor version numbers are necessary: for example, the .NET Core SDK version 2.1.300 can be referenced as the package `dotnet-sdk-2.1`. Prior versions require the en ...
```
This should be "dotnet-sdk-2.1.200"

JRAlexander reviewed (changes requested):
Looks good, @BillWagner! Just a few small changes and then you should be ready to rock.

BillWagner committed:
respond to content feedback.

JRAlexander reviewed (changes requested):
Looks good, one more small spelling thing and we should be good to go!

BillWagner committed:
typo

JRAlexander commented:
Looks great, @BillWagner! :shipit: at your convenience!!!

BillWagner committed:
branding: macOS is the official capitalization

BillWagner committed:
final proofread

mairaw commented on docs/core/tutorials/metapackages.md:
```diff
@@ -0,0 +1,14 @@
+
```
this is missing metadata

BillWagner replied:
This file was supposed to be removed (The information from it is now in the versions/index.md file.)

BillWagner committed:
Final updates

There are three updates in this commit:

1. Remove the metapackages.md file. It should have been removed earlier.
2. Remove the version history table, and link to the .NET Core downloads page. This page has been updated to include all the relevant information, and is updated with each release. Also, add a link to the archive page for information on older releases.
3. Update the net-standard-table to use Framework 4.7.2 for .NET Standard, and link to the download page for .NET Core 2.0 tooling for Visual Studio 2015.

BillWagner commented:
@mairaw Can you give the last commit one final review? It's larger than I'd like to merge without another set of :eyes:

BillWagner commented:
I think this got caught between PR merges. closing and reopening to build.

mairaw commented:
Will do @BillWagner!

mairaw reviewed (approved):
@BillWagner this looks great! Thanks for taking this on. I did leave some comments for you to consider before merging.

BillWagner commented on includes/net-standard-table.md:
```diff
@@ -1,7 +1,7 @@
 |----------------------------|-------|--------|-------|-------|-------|------------|------------|------------|
 | .NET Core                  | 1.0   | 1.0    | 1.0   | 1.0   | 1.0   | 1.0        | 1.0        | 2.0        |
-| .NET Framework <sup>1</sup>| 4.5   | 4.5    | 4.5.1 | 4.6   | 4.6.1 | 4.6.1      | 4.6.1      | 4.6.1      |
+| .NET Framework <sup>1</sup>| 4.5   | 4.5    | 4.5.1 | 4.6   | 4.6.1 | 4.6.1      | 4.6.1      | 4.7.2      |
```
I was going by this comment: https://github.com/dotnet/docs/issues/5915#issuecomment-406438591   @terrajobst Thoughts? 4.7.2 came out with the April Window 10 update.

BillWagner replied:
Dropped #5915 from the list of issues addressed by this PR, and reverted this change.

BillWagner committed:
respond to feedback

BillWagner committed:
revert framework cell for .NET standard 2.0
